src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,43 +481,110 @@ public Authentication authenticate(Authentication authentication) throws Authent
+                        	newAuthMgr = new GlobalMatrixAuthorizationStrategy();
+                        }
+                        
+                        // we add the latest permissions instead of copy if a user from openshift (not predefined user), and we are logging in as that user
+                        // we copy the permissions if it is a user from openshift that is not the user we are logging in as, or it is one of the predefined users 
+                    	//TODO though long term, we may want to purge admin/system_builder/view
+                        boolean addInsteadOfCopy = !classicDefaultJenkinsUsers.contains(info.getName());
+                        if (LOGGER.isLoggable(Level.FINE))
+                        	LOGGER.fine(String.format("onSuccess: got users %s and permissions %s addInsteadOfCopy %s for user %s", usersGroups.toString(), permissionGroups.toString(), Boolean.toString(addInsteadOfCopy), info.getName()));
+                        for (String userGroup : usersGroups) {                    	
+                        	// copy any of the user's permissions from the prior auth mgr to our new one
+                        	for (PermissionGroup pg : permissionGroups) {
+                        		for (Permission p : pg.getPermissions()) {
+                        			if (existingAuthMgr.hasPermission(userGroup, p)) {
+                        				newAuthMgr.add(p, userGroup);
+                        			}
+                        		}
+                        	}
+                        }
+                        
+                    	if (addInsteadOfCopy) {
+                        	if (LOGGER.isLoggable(Level.FINE))
+                        		LOGGER.fine(String.format("onSuccess: adding permissions to new user %s based on openshift roles %s", info.getName(), allowedRoles));
+                        	
+                        	// map OpenShift user based on role to Jenkins user with analogous permissions
+                        	//TODO can we minimally assume view role ?  assuming so while waiting for the build.oenshift.io.jenkins related changes we think are coming
+                        	// and see if a self-sar against the view role comes back and says allowed (currently returns allowed as false for the view role)
+                        	newAuthMgr.add(Hudson.READ, info.getName());
+                        	newAuthMgr.add(Item.READ, info.getName());
+                			if (allowedRoles.contains("edit") || allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Item.BUILD, info.getName());
+                				newAuthMgr.add(Item.CONFIGURE, info.getName());
+                				newAuthMgr.add(Item.CREATE, info.getName());
+                				newAuthMgr.add(Item.DELETE, info.getName());
+                				newAuthMgr.add(Item.WORKSPACE, info.getName());
+                				newAuthMgr.add(SCM.TAG, info.getName());
+                			}
+                			if (allowedRoles.contains("admin")) {
```

Comment:
> I'm assuming (per the comment above with the TODO / question) that everyone
> will have at least "view" access.
> Please correct me if that is not the case.  If it is not the case, the next
> step will be to confirm what is needed to at least
> log in successfully (minimally need Hudson.READ for sure; will have to try
> with and without Item.READ) vs. what
> else might we want to add that would be analogous for the "view" role in
> OpenShift.
> 
> The "view" user predefined in our Jenkins image today has Hudson.READ and
> Item.READ.

No.  I'll work up a pull to create the role to add jenkins permissions (based on these values), but most users will have no rights to a random jenkins instance.